Sources/_StringProcessing/Unicode/ASCII.swift

```diff
@@ -85,11 +85,14 @@ extension String {
   /// and we can give the right `next` index, not requiring the caller to re-adjust it
   /// TODO: detailed description of nuanced semantics
   func _quickASCIICharacter(
-    at idx: Index
+    at idx: Index,
+    limitedBy end: Index
   ) -> (first: UInt8, next: Index, crLF: Bool)? {
     // TODO: fastUTF8 version
-
-    if idx == endIndex {
+    assert(String.Index(idx, within: unicodeScalars) != nil)
+    assert(idx <= end)
+    
+    if idx == end {
```

Comment:
Post-merge cleanup: it seems like anything that's not permissive of an empty match should fail long before reaching this code